src/interp/interp.cc

```diff
@@ -260,31 +264,44 @@ void Store::Collect() {
     thread->Mark();
   }
 
-  // TODO: better GC algo.
-  // Loop through all newly marked objects and mark their referents.
-  std::vector<bool> all_marked(object_count, false);
-  bool new_marked;
-  do {
-    new_marked = false;
-    for (size_t i = 0; i < object_count; ++i) {
-      if (!all_marked[i] && marks_[i]) {
-        all_marked[i] = true;
-        objects_.Get(i)->Mark(*this);
-        new_marked = true;
-      }
-    }
-  } while (new_marked);
+  // This vector is often empty since the default maximum
+  // recursion is usually enough to mark all objects.
+  while (!gc_context_.untraced_objects.empty()) {
```

Comment:
May I add some review?
It seems that wabt adopts UNLIKELY macro (`WABT_UNLIKELY`), so how about using it for this uncommon condition?

Reply:
I have added these macros.